server.js

```diff
@@ -256,6 +256,7 @@ app.prepare().then(async () => {
     const start = req.query.start ? new Date(req.query.start) : new Date(0) // start of EPOCH
     const end = req.query.end ? new Date(req.query.end) : new Date() // now
     const limit = req.query.limit ? `limit ${parseInt(req.query.limit)}` : ""
+    const app = req.query.app ? `WHERE entity_id = '${req.query.app}'` : ""
```

Comment:
If I read it correctly, it introduces SQL injection here. Even though the database does not have sensitive data and is already public, I am not a fun of letting this practice passed easily.

Reply:
It does use SQL injection. This shouldn't be a concern due to the readonly nature of the credentials, however i have now added parsing to remove special characters and only accept a single word to negate any SQL injection attempts.